src/main/java/org/hmhb/authentication/AuthenticationController.java

```diff
@@ -0,0 +1,45 @@
+package org.hmhb.authentication;
+
+import javax.annotation.Nonnull;
+
+import com.codahale.metrics.annotation.Timed;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import static java.util.Objects.requireNonNull;
+
+@RestController
+public class AuthenticationController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationController.class);
+
+    private final AuthenticationService service;
+
+    @Autowired
+    public AuthenticationController(
+            @Nonnull AuthenticationService service
+    ) {
+        this.service = requireNonNull(service, "service cannot be null");
+    }
+
+    @Timed
+    @RequestMapping(
+            method = RequestMethod.POST,
+            produces = MediaType.APPLICATION_JSON_VALUE,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            path = "/auth/google"
```

Comment:
This is what the satellizer widget calls into after getting a request token from google.  It then passes me clientId, redirectUri, and code.  I then use that information to call into google to retrieve the user's email.